### PR TITLE
[Feature]: Add C Language Course 

### LINKED ISSUE #3715: [Feature]: Add `C Language Course`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

C Intro:
- C is a general-purpose, procedural programming language that was developed in the early 1970s by Dennis Ritchie at Bell Labs

Real world Uses:

- **Operating Systems:** Many operating systems, including UNIX, Linux, and Windows, are written in C due to its performance and low-level capabilities.
- **Embedded Systems:** C is widely used in embedded systems programming due to its ability to interact directly with hardware.
- **Game Development:** Game engines and real-time applications often use C or C++ for their performance and control over system resources.
- **Compilers and Interpreters:** Many compilers and interpreters for other programming languages are written in C.

### Use Case

@Ajay-Dhangar ,please assign me...

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by sivaprasath2004**
Hello @Ajay-Dhangar ,

## Related Issue

Fixes #3715 

## Description

[Feature]: Add C Language Course 

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Screencast from 2024-07-22 14-06-24.webm](https://github.com/user-attachments/assets/c904378d-a969-48bd-8fdd-82418afb4420)

Screenshot from 2024-07-22 14-13-26
 
Screenshot from 2024-07-22 14-13-44
 
Screenshot from 2024-07-22 14-19-29

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [ ] I have read the resources for guidance listed below.
- [ ] I have followed security best practices in my code changes.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [x] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [ ] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [ ] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [x] [Markdown Guide](https://www.markdownguide.org/)
- [ ] [MDX Documentation](https://mdxjs.com/docs/)
- [ ] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [x] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [ ] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [ ] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

File: src/data/courses/index.tsx [hunk 1]
```diff
@@ -344,8 +344,17 @@ const courses = [
       imageUrl: "/img/svg/developer_activity.svg",
       author: "Ajay Dhangar",
       link: "/courses/category/nextjs"
+    },
+    {
+      "id": 40,
+      "title": "C Language",
+      "description": "This course is for absolute beginners who want to learn C Language.",
+      "category": "figma",
```
Comment (Ajay-Dhangar) [inline src/data/courses/index.tsx:352]: change `"category": "figma",`

File: courses/C/intermediate-level/File-Handling/intro.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+| Close the file using     |
+| fclose                   |
++--------------------------+
+        |
+        v
++--------------------------+
+| End                      |
++--------------------------+
+```
+
```
Comment (Ajay-Dhangar) [inline courses/C/intermediate-level/File-Handling/intro.md:67]: You are working on this pull request to add a new C language course to CodeHarborHub. You need Mermaid diagrams to create the flowchart for file handling, which helps visualize the flow of file operations and error handling. Here are the links to the documentation and an example of how I used Mermaid:

* [Mermaid documentation](https://mermaid.js.org/intro/)
* [Diagram with Mermaid](https://codeharborhub.github.io/community/features#diagrams-with-mermaid)

File: courses/C/intermediate-level/Error-handling/types.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+ | Use perror to print error |
+ | message and handle error  |
+ +----------------------------+
+        |
+        v
+ +----------------------------+
+ | End                        |
+ +----------------------------+
+```
+
```
Comment (Ajay-Dhangar) [inline courses/C/intermediate-level/Error-handling/types.md:100]: You are working on this pull request to add a new C language course to CodeHarborHub. You need Mermaid diagrams to create the flowchart for file handling, which helps visualize the flow of file operations and error handling. Here are the links to the documentation and an example of how I used Mermaid:

* [Mermaid documentation](https://mermaid.js.org/intro/)
* [Diagram with Mermaid](https://codeharborhub.github.io/community/features#diagrams-with-mermaid)

File: courses/C/intermediate-level/Dynamic-memory-allocation/Intro.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+| Free the allocated memory|
+| using free               |
++--------------------------+
+        |
+        v
++--------------------------+
+| End                      |
++--------------------------+
+```
+
```
Comment (Ajay-Dhangar) [inline courses/C/intermediate-level/Dynamic-memory-allocation/Intro.md:66]: You are working on this pull request to add a new C language course to CodeHarborHub. You need Mermaid diagrams to create the flowchart for file handling, which helps visualize the flow of file operations and error handling. Here are the links to the documentation and an example of how I used Mermaid:

* [Mermaid documentation](https://mermaid.js.org/intro/)
* [Diagram with Mermaid](https://codeharborhub.github.io/community/features#diagrams-with-mermaid)

File: courses/C/intermediate-level/Advanced pointers/dynamic-array-pointer.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
++--------------------------+
+| Free the allocated memory|
+| using free               |
++--------------------------+
+        |
+        v
++--------------------------+
+| End                      |
++--------------------------+
+```
```
Comment (Ajay-Dhangar) [inline courses/C/intermediate-level/Advanced pointers/dynamic-array-pointer.md:138]: Use a mermaid diagram for flow chart for example https://mermaid.js.org/syntax/flowchart.html#minimum-length-of-a-link

File: courses/C/intermediate-level/ Preprocessor Directives/intro.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+ | Check additional         |
+ | conditions with #elif,   |
+ | #else if applicable     |
+ +--------------------------+
+        |
+        v
++--------------------------+
+ | End                      |
+ +--------------------------+
+```
```
Comment (Ajay-Dhangar) [inline courses/C/intermediate-level/ Preprocessor Directives/intro.md:60]: Use a mermaid diagram for flow chart for example https://mermaid.js.org/syntax/flowchart.html#minimum-length-of-a-link

File: courses/C/beginner-level/Functions/Scope-of-variable.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+            v
+  +-------------------------+
+  |    Function Return      |
+  +-------------------------+
+            |
+            v
+  +-------------------------+
+  |    Continue Execution   |
+  +-------------------------+
+```
```
Comment (Ajay-Dhangar) [inline courses/C/beginner-level/Functions/Scope-of-variable.md:121]: Use a mermaid diagram for a flow chart for example https://mermaid.js.org/syntax/flowchart.html

**Review by Ajay-Dhangar — COMMENTED** [review]
@sivaprasath2004, I think you should use Mermaid diagrams in your code. Our site fully supports Mermaid diagrams, so you can use them to easily define your content. To get started, you can check out the official Mermaid docs at https://mermaid.js.org/syntax/flowchart.html. Try writing some flowcharts and diagrams. Also, importantly, make sure that each folder or directory and its files have unique IDs, or we'll get duplicate routes warnings or errors.

**Comment by sivaprasath2004** [comment]
Hello @Ajay-Dhangar changed..


Screenshot from 2024-07-22 17-48-58

**Comment by sivaprasath2004** [comment]
Screenshot from 2024-07-22 18-02-24

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**